orchestra/frontend/src/components/TasksList/TasksList.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useState, useEffect } from 'react'
+import {
+  useHistory
+} from 'react-router-dom'
+
+import './TasksList.scss'
+
+import Table from '@b12/metronome/components/layout/table/Table.es6'
+import TableHead from '@b12/metronome/components/layout/table/TableHead.es6'
+import TableBody from '@b12/metronome/components/layout/table/TableBody.es6'
+import TableRow from '@b12/metronome/components/layout/table/TableRow.es6'
+import TableCell from '@b12/metronome/components/layout/table/TableCell.es6'
+import TextField from '@b12/metronome/components/form/textfield/TextField.es6.js'
+import Badge from '@b12/metronome/components/layout/badge/Badge.es6.js'
+import StatusIndicator from '@b12/metronome/components/layout/status-indicator/StatusIndicator.es6.js'
+import {
+  CaretDown
+} from '@b12/metronome/components/Icons.es6.js'
+
+import AnimatedCircle from '../../assets/AnimatedCircle'
+
+import { getPrettyDatetime, specialFormatIfToday, isOutdated } from '../../util/time'
+import { Task } from 'state/slices/dashboardTasks'
+
+type ProjectListProps = {
+  status: any;
+  tasks: Task[];
+  isLoading?: boolean;
+}
+
+type SortStatusType = {
+  column: string; // enum of rows
+  direction: 'ascending' | 'descending';
+}
+
+const TaskList = ({ status, tasks, isLoading = false }: ProjectListProps) => {
+  const [sortStatus, setSortStatus] = useState({
+    column: null,
+    direction: 'ascending'
+  })
+  const [sortedTasks, setSortedTasks] = useState<Task[]>([])
+
+  useEffect(() => {
+    setSortedTasks(tasks)
+  }, [tasks])
+
+  useEffect(() => {
+    // TODO: some small buggo that doesn't sort upon first click of sortStatus.
```

Comment:
easy to resolve?

Reply:
I believe you resolved this now, right?

Reply:
Sorry I missed this the first time round! I've just redid the logic and abstracted it out into a helper for clarity.